### DISCUSSION TITLE
Update text without overwrite old text...

### DISCUSSION THREAD

**Discussion by fsadak** [Bodmer/TFT_eSPI#2928] [Q&A - General]
In the graphical user interface I made, I added all the basic visuals in a single image file and uploaded it to the screen as a background image, and then I send the variable values to the screen by writing them in the code. However, when a value on the screen changes, it overwrites the old value and becomes unreadable. I came across a similar problem while looking through the discussion threads, but there the solution was suggested by changing the background color of the text. In my case this doesn't work because there is already a background color where the text is written, so I don't use a background color in the text.

I would like to know if there is another way to overcome this problem.

Thank you...

**Comment by Bodmer** [comment]
A few potential ways:

1. create a small sprite that can contain the value and draw the background in it with the right -x, -y offset. Print the value in the sprite then paste the srite to the screen at x,y
2. A faster way if the value position is static is to have 2 sprites, one with a copy of the small area of the background where the value will be printed, copy that to a second  sprite, print the value in the second sprite, then push the 2nd sprite to the screen
